qiskit_machine_learning/datasets/entanglement_concentration.py

```diff
@@ -0,0 +1,361 @@
+# This code is part of a Qiskit project.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Entanglement Concentration
+"""
+from __future__ import annotations
+
+import warnings
+import os
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.quantum_info import Operator, Statevector
+from qiskit.circuit import ParameterVector
+
+from ..utils import algorithm_globals
+
+
+# pylint: disable=too-many-positional-arguments
+def entanglement_concentration_data(
+    training_size: int,
+    test_size: int,
+    n: int,
+    mode: str = "easy",
```

Comment:
My take, instead of suppressing the warning, would be to add `*,` after the `n` parameter which would allow the first 3 arguments to be positional but require the following, which do have defaults so they do not need to be provided, be provided when doing so as keyword arguments only.